src/room-terrain.ts

```diff
@@ -9,6 +9,12 @@ interface RoomTerrain {
      * @return number Number of terrain mask like: TERRAIN_MASK_SWAMP | TERRAIN_MASK_WALL
      */
     get(x: number, y: number): 0 | TERRAIN_MASK_WALL | TERRAIN_MASK_SWAMP;
+    /**
+     * Get copy of underlying static terrain buffer.
+     * @param destinationArray (optional) A typed array view in which terrain will be copied to.
+     * @return Uint8Array Copy of underlying room terrain as a new Uint8Array typed array of size 2500.
+     */
+    getRawBuffer(destinationArray?: Uint8Array): Uint8Array;
```

Comment:
We should define a type alias of the `underlying representation`, then use it as the default type parameter.

Here are my suggestions for typing this method. 
The first seems to be less correct. The second uses overloads, but I think it fits better.
```ts
getRawBuffer<T extends TypedArray = Uint8Array>(destinationArray?: T): T;
```
or 
```ts
getRawBuffer<T extends TypedArray>(destinationArray: T): T;
getRawBuffer(): Uint8Array;
```

In case there is no built-in union for TypedArray : 
```ts
type TypedArray = Int8Array | Uint8Array | Int16Array | Uint16Array | Int32Array | Uint32Array | Uint8ClampedArray | Float32Array | Float64Array;
```